prisma/schema.prisma

```diff
@@ -25,6 +25,23 @@ model User {
     gift      Gift[]
     Session   Session[]
     Feedback  Feedback[]
+    Person    Person[]
+}
+
+model Person {
+    id            Int            @id @default(autoincrement())
+    uuid          String         @unique @default(uuid())
+    name          String
+    sendReminders Boolean
+    // map:in käyttö napattu täältä https://github.com/prisma/prisma/discussions/11108
+    // onko väärä yhdistää kuvan "omistaja" userUUID:llä?
+    pictureId     ProfilePicture @relation(fields: [userUUID], references: [userUUID], onDelete: Cascade, map: "picture_fk")
+    userUUID      String         @unique
+    // onDelete: Cascade pitäisi olla ihan ok idea, koska jos User poistetaan ei Person:ia pitäisi tarvita
+    userId        User           @relation(fields: [userUUID], references: [uuid], onDelete: Cascade, map: "user_fk")
+    createdAt     DateTime       @default(now())
+    updatedAt     DateTime       @updatedAt
+    Anniversary   Anniversary[]
 }
```

Comment:
Lisäsin rivit, mitkä olit kirjoittanut #55 issueen. Lisäsin myös oman rivin `userUUID` (lisäsin myös noihin kahteen muuhun), jonka tarkoitus on yhdistää käyttäjä. Ja kun käyttäjä on yhdistetty kätettäisiin sitä esim. kuvan omistajan yhdistämiseen. Onko huono idea, onko tähän joku näppärämpi? :thinking: 

Merkkasin nyt sekä `picutreId`n ja `userId`n käyttämään `userUUID`:tä relaatioissa. Tässä muodostuu ongelma, koska käytetään samaa kenttää (`userUUID`) useammassa relaatiossa. Löysin ohjeen: https://www.github.com/prisma/prisma/discussions/11108, jossa neuvottiin käyttämään `map`ia. Onko tämä ihan ok? :smile: 

Tämä seuraava kohta epäilyttää hiukan :smile: 

`pictureId`lle on asetettu `onDelete: Cascade`. Ymmärykseni mukaan tämä on `1-1` eli `OnDelete` ilmeisesti toimii niin: jos `ProfilePicture` poistetaan poistuu `Person` samalla joka ei ole toivottu lopputulos. Toivottu lopputulos lienee se, että jos `Person` poistetaan, poistetaan samalla `ProfilePicture` (tässä voin mennä mettään ja pahasti :rofl:)

`userId`:lle on asetettu kanssa `onDelete: Cascade`. Prisman ohjeiden mukaan tämä on `1-n`, koska rivillä 28 on asetettu: `Person    Person[]`. Tämä `OnDelete` pitäisi seurata `User`ia niin, että jos `User` poistetaan, poistetaan samalla `Person`. Mutta ei toisinpäin, eli jos `Person` poistetaan ei poisteta `User`ia. Onko tämä ok vai onko tässä jokin vaara, jota näin aloittelijana en huomaa :smile: 

Lopussa vielä `Anniversary`, joka pitäisi olla `1-n`. Yhdellä henkilöllä voi olla monta merkkipäivää asetettuna

Reply:
> Lisäsin rivit, mitkä olit kirjoittanut #55 issueen. Lisäsin myös oman rivin `userUUID` (lisäsin myös noihin kahteen muuhun), jonka tarkoitus on yhdistää käyttäjä. Ja kun käyttäjä on yhdistetty kätettäisiin sitä esim. kuvan omistajan yhdistämiseen. Onko huono idea, onko tähän joku näppärämpi? 🤔

`userId` kenttä pitäisi oikeastaan olla `user` jos tehdään kuten esim. Feedback-modelissa. ProfilePicture oli tarkoitus olla Person-modelin kuva eikä userin, mutta nimi on kieltämättä hieman hämäävä. Tehdään siis kaksi muutosta:
- Uudelleennimetään `ProfilePicture` -> `PersonPicture`.
- Siirretään `pictureId` foreign key PersonPicture-modeliin (1-to-1 relaatioissa on käytäntö laittaa foreign key siihen modeliin joka ei voi esiityä ilman toista. Nyt PersonPicturea ei voi olla ilman Person-modelia, mutta toisin päin voi olla).

> Merkkasin nyt sekä `picutreId`n ja `userId`n käyttämään `userUUID`:tä relaatioissa. Tässä muodostuu ongelma, koska käytetään samaa kenttää (`userUUID`) useammassa relaatiossa. Löysin ohjeen: https://www.github.com/prisma/prisma/discussions/11108, jossa neuvottiin käyttämään `map`ia. Onko tämä ihan ok? 😄

Edellä olevien muutosten jälkeen ei pitäisi olla tarvetta `map`:lle.

> Tämä seuraava kohta epäilyttää hiukan 😄
> 
> `pictureId`lle on asetettu `onDelete: Cascade`. Ymmärykseni mukaan tämä on `1-1` eli `OnDelete` ilmeisesti toimii niin: jos `ProfilePicture` poistetaan poistuu `Person` samalla joka ei ole toivottu lopputulos. Toivottu lopputulos lienee se, että jos `Person` poistetaan, poistetaan samalla `ProfilePicture` (tässä voin mennä mettään ja pahasti 🤣)

Toimii toivotulla tavalla. `onDelete: Cascade` asetetaan aina siihen modeliin johon laitetaan foreign key, ja tarkoittaa sitä että jos model mihin foreign key osoittaa poistetaan niin silloin myös viittaava malli poistuu. [Prisman dokumentaatiossa](https://www.prisma.io/docs/orm/prisma-schema/data-model/relations/referential-actions) on tästä lisää.
 
> `userId`:lle on asetettu kanssa `onDelete: Cascade`. Prisman ohjeiden mukaan tämä on `1-n`, koska rivillä 28 on asetettu: `Person Person[]`. Tämä `OnDelete` pitäisi seurata `User`ia niin, että jos `User` poistetaan, poistetaan samalla `Person`. Mutta ei toisinpäin, eli jos `Person` poistetaan ei poisteta `User`ia. Onko tämä ok vai onko tässä jokin vaara, jota näin aloittelijana en huomaa 😄

Just noin siinä käy, eli jos `User` poistetaan niin kaikki `Person`it poistetaan automaattisesti mutta ei toisin päin.
 
> Lopussa vielä `Anniversary`, joka pitäisi olla `1-n`. Yhdellä henkilöllä voi olla monta merkkipäivää asetettuna

 Jep!